.github/workflows/ci.yml

```diff
@@ -14,156 +13,128 @@ on:
 env:
   IMAGE_NAME: maykinmedia/open-product
   DJANGO_SETTINGS_MODULE: openproduct.conf.ci
-  DOCKER_BUILDKIT: '1'
+  DB_PASSWORD: ''
+  DB_USER: postgres
 
 jobs:
-
-  setup:
-    name: Set up the build variables
-    runs-on: ubuntu-latest
-    outputs:
-      tag: ${{ steps.vars.outputs.tag }}
-      git_hash: ${{ steps.vars.outputs.git_hash }}
-
+  # determine changed files to decide if certain jobs can be skipped or not
+  changed-files:
+    runs-on: ubuntu-latest  # windows-latest | macos-latest
+    name: Determine changed files
     steps:
-      - name: Extract version information
-        id: vars
-        run: |
-          # Strip git ref prefix from version
-          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
 
-          # Strip "v" prefix from tag name (if present at all)
-          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-          # Use Docker `latest` tag convention
-          [ "$VERSION" == "main" ] && VERSION=latest
+      - name: Get changed PY files
+        id: changed-py-files
+        run: bin/check_changed_files.sh ^src/.*\.py$
```

Comment:
Since you use `paths` elsewhere as well, is it possible to replace this script stuff with `paths`?

Reply:
the changed-files output is only used for the test jobs so i could have a separate file for those. But i feel like we should keep it consistent with the other repositories.

Reply:
Ah nvm, checked it and `paths` can only be used at the workflow level and not at the job level apparently, so let's leave it as it is